--- a/kioslave/audiocd/plugins/wav/encodercda.cpp	(revision 865239)
+++ b/kioslave/audiocd/plugins/wav/encodercda.cpp	(revision 865240)
@@ -42,7 +42,10 @@
 // Remove this by calculating CD_FRAMESIZE_RAW from the frames
 extern "C"
 {
+  //cdda_interface.h in cdparanoia 10.2 has a member called 'private' which the C++ compiler doesn't like
+  #define private _private
   #include <cdda_interface.h>
+  #undef private
 }
 
 inline int16_t swap16 (int16_t i)
--- a/kioslave/audiocd/audiocd.cpp	(revision 865239)
+++ b/kioslave/audiocd/audiocd.cpp	(revision 865240)
@@ -27,8 +27,11 @@
 #include <kdemacros.h>
 extern "C"
 {
+	//cdda_interface.h in cdparanoia 10.2 has a member called 'private' which the C++ compiler doesn't like
+	#define private _private   
 	#include <cdda_interface.h>
 	#include <cdda_paranoia.h>
+	#undef private
 	void paranoiaCallback(long, int);
 
 	KDE_EXPORT int kdemain(int argc, char ** argv);
