--- kdemultimedia-4.0.71+svn801741.orig/kmix/CMakeLists.txt	2008-04-28 08:59:35.000000000 +0000
+++ kdemultimedia-4.0.71+svn801741/kmix/CMakeLists.txt	2008-04-28 09:00:32.000000000 +0000
@@ -50,7 +50,7 @@
 
 kde4_add_kdeinit_executable( kmix ${kmix_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kmix ${KDE4_KUTILS_LIBS} ${KDE4_PHONON_LIBS} ${KDE4_SOLID_LIBS})
+target_link_libraries(kdeinit_kmix ${KDE4_KUTILS_LIBS} ${KDE4_PHONON_LIBS} ${KDE4_SOLID_LIBS} ${QT_QTXML_LIBRARY})
 
 if (HAVE_LIBASOUND2)
     target_link_libraries(kdeinit_kmix ${ASOUND_LIBRARY})
@@ -78,7 +78,7 @@
 
 kde4_add_kdeinit_executable( kmixctrl ${kmixctrl_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kmixctrl ${KDE4_KDEUI_LIBS} ${KDE4_SOLID_LIBS})
+target_link_libraries(kdeinit_kmixctrl ${KDE4_KDEUI_LIBS} ${KDE4_SOLID_LIBS} ${QT_QTDBUS_LIBRARY} ${QT_QTXML_LIBRARY})
 
 if (HAVE_LIBASOUND2)
     target_link_libraries(kdeinit_kmixctrl ${ASOUND_LIBRARY})
--- kdemultimedia-4.0.71+svn801741.orig/libkcddb/CMakeLists.txt	2008-04-28 08:22:51.000000000 +0000
+++ kdemultimedia-4.0.71+svn801741/libkcddb/CMakeLists.txt	2008-04-28 08:22:51.000000000 +0000
@@ -60,7 +60,7 @@
 
 kde4_add_library(kcddb SHARED ${kcddb_LIB_SRCS})
 
-target_link_libraries(kcddb ${KDE4_KDE3SUPPORT_LIBS})
+target_link_libraries(kcddb ${KDE4_KDE3SUPPORT_LIBS} ${QT_QTDBUS_LIBRARY} ${QT_QTNETWORK_LIBRARY} ${QT_QTXML_LIBRARY})
 if(MUSICBRAINZ_FOUND)
     target_link_libraries(kcddb ${MUSICBRAINZ_LIBRARIES})
 endif(MUSICBRAINZ_FOUND)
--- kdemultimedia-4.0.71+svn801741.orig/libkcompactdisc/CMakeLists.txt	2008-04-28 08:38:57.000000000 +0000
+++ kdemultimedia-4.0.71+svn801741/libkcompactdisc/CMakeLists.txt	2008-04-28 08:50:33.000000000 +0000
@@ -47,7 +47,7 @@
 
 kde4_add_library(kcompactdisc SHARED ${kcompactdisc_LIB_SRCS})
 
-target_link_libraries(kcompactdisc ${KDE4_KIO_LIBS} ${KDE4_PHONON_LIBS})
+target_link_libraries(kcompactdisc ${KDE4_KIO_LIBS} ${KDE4_PHONON_LIBS} ${KDE4_SOLID_LIBRARY} ${CMAKE_THREAD_LIBS_INIT})
 if (HAVE_LIBASOUND2)
     target_link_libraries(kcompactdisc ${ASOUND_LIBRARY})
 endif (HAVE_LIBASOUND2)
