--- a/kmix/CMakeLists.txt
+++ b/kmix/CMakeLists.txt
@@ -50,7 +50,7 @@
 
 kde4_add_kdeinit_executable( kmix ${kmix_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kmix ${KDE4_KUTILS_LIBS} ${KDE4_PHONON_LIBS} ${KDE4_SOLID_LIBS})
+target_link_libraries(kdeinit_kmix ${KDE4_KUTILS_LIBS} ${KDE4_PHONON_LIBS} ${KDE4_SOLID_LIBS} ${QT_QTXML_LIBRARY})
 
 if (HAVE_LIBASOUND2)
     target_link_libraries(kdeinit_kmix ${ASOUND_LIBRARY})
@@ -78,7 +78,7 @@
 
 kde4_add_kdeinit_executable( kmixctrl ${kmixctrl_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kmixctrl ${KDE4_KDEUI_LIBS} ${KDE4_SOLID_LIBS})
+target_link_libraries(kdeinit_kmixctrl ${KDE4_KDEUI_LIBS} ${KDE4_SOLID_LIBS} ${QT_QTDBUS_LIBRARY} ${QT_QTXML_LIBRARY})
 
 if (HAVE_LIBASOUND2)
     target_link_libraries(kdeinit_kmixctrl ${ASOUND_LIBRARY})
--- a/libkcddb/CMakeLists.txt
+++ b/libkcddb/CMakeLists.txt
@@ -60,7 +60,7 @@
 
 kde4_add_library(kcddb SHARED ${kcddb_LIB_SRCS})
 
-target_link_libraries(kcddb ${KDE4_KDE3SUPPORT_LIBS})
+target_link_libraries(kcddb ${KDE4_KDE3SUPPORT_LIBS} ${QT_QTDBUS_LIBRARY} ${QT_QTNETWORK_LIBRARY} ${QT_QTXML_LIBRARY})
 if(MUSICBRAINZ_FOUND)
     target_link_libraries(kcddb ${MUSICBRAINZ_LIBRARIES})
 endif(MUSICBRAINZ_FOUND)
--- a/libkcompactdisc/CMakeLists.txt
+++ b/libkcompactdisc/CMakeLists.txt
@@ -47,7 +47,7 @@
 
 kde4_add_library(kcompactdisc SHARED ${kcompactdisc_LIB_SRCS})
 
-target_link_libraries(kcompactdisc ${KDE4_KIO_LIBS} ${KDE4_PHONON_LIBS})
+target_link_libraries(kcompactdisc ${KDE4_KIO_LIBS} ${KDE4_PHONON_LIBS} ${KDE4_SOLID_LIBRARY} ${CMAKE_THREAD_LIBS_INIT})
 if (HAVE_LIBASOUND2)
     target_link_libraries(kcompactdisc ${ASOUND_LIBRARY})
 endif (HAVE_LIBASOUND2)
--- a/dragonplayer/src/app/CMakeLists.txt
+++ b/dragonplayer/src/app/CMakeLists.txt
@@ -32,6 +32,8 @@
 target_link_libraries(dragon
     ${KDE4_KUTILS_LIBS}
     ${KDE4_PHONON_LIBS}
+    ${KDE4_SOLID_LIBS}
+    ${KDE4_KIO_LIBRARY}
     ${XINE_LIBRARY} )
 
 install(TARGETS dragon DESTINATION ${BIN_INSTALL_DIR})
@@ -44,6 +46,7 @@
 kde4_add_plugin(dragonpart ${videoplayer_part_SRCS})
 
 target_link_libraries(dragonpart
+${KDE4_SOLID_LIBS}
     ${KDE4_KPARTS_LIBS}
     ${KDE4_PHONON_LIBS}
     ${XINE_LIBRARY} )
